moz-webgpu-cts/src/report.rs

```diff
@@ -0,0 +1,111 @@
+use serde::{
+    de::{Deserializer, Error},
+    Deserialize,
+};
+
+use crate::metadata::{BuildProfile, Platform, SubtestOutcome, TestOutcome};
+
+#[derive(Debug, Deserialize)]
+pub(crate) struct ExecutionReport {
+    pub run_info: RunInfo,
+    #[serde(rename = "results")]
+    pub entries: Vec<TestExecutionEntry>,
+}
+
+#[derive(Debug)]
+pub(crate) struct RunInfo {
+    pub platform: Platform,
+    pub build_profile: BuildProfile,
+}
+
+impl<'de> Deserialize<'de> for RunInfo {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        #[derive(Debug, Deserialize)]
+        struct ActualRunInfo {
+            os: String,
+            processor: String,
+            win11_2009: bool,
+            debug: bool,
+        }
+
+        let ActualRunInfo {
+            os,
+            processor,
+            win11_2009,
+            debug,
+        } = ActualRunInfo::deserialize(deserializer)?;
+
+        let platform = match &*os {
+            "win" => {
+                if processor == "x86_64" && win11_2009 {
+                    Platform::Windows
+                } else {
+                    return Err(D::Error::custom("asdf"));
+                }
+            }
+            // TODO: Make sure we're not missing any important fields to validate for
+            // Mac
+            "mac" => Platform::MacOs,
+            // TODO: Make sure we're not missing any important fields to validate for
+            // Linux
```

Comment:
Will break these out into separate issues.

Reply:
Broken out to https://github.com/ErichDonGubler/moz-webgpu-cts/issues/54, https://github.com/ErichDonGubler/moz-webgpu-cts/issues/55. Fixed up with fe21153.